params.go

```diff
@@ -0,0 +1,17 @@
+package dnscache
+
+type Param struct {
+	apply func(r *Resolver)
+}
+
+func WithRefreshCompletedListener(listener func()) Param {
+	return Param{apply: func(r *Resolver) {
+		r.onCacheRefreshedFn = listener
+	}}
+}
+
+func WithCustomIPLookupFunc(fn LookupIPFn) Param {
+	return Param{apply: func(r *Resolver) {
+		r.ipLookupFn = fn
+	}}
+}
```

Comment:
These are currently only used internally for testing, is it useful to be exporting these?

Reply:
Yep, did not like them as well. Rewritten tests so we:
- would not need refreshListener
- would not expose lookupFn